static_resources/markdown-editor.js

```diff
@@ -119,45 +135,102 @@ function length(str) {
                     start: '1. ', end: '', placeholder: 'List Item',
                     line: true, block: true
                 },
+                break: {
+                    start: '', end: '\n{.clearfix}', placeholder: '',
+                    line: true, block: true
+                },
+                custom: {
+                    start: '', end: ' {.custom-class-here}', placeholder: '',
+                    line: true, block: true
```

Comment:
I also think removing the space before the end will make it more useful (as there can't be a space between (eg) images and the `{`. On the other hand, the space is needed for headers, but paragraphs need a newline rather than either. Its a bit of a mess, so any option is going to be a compromise over all the others. At least no space means they can add in the others, I guess?